initialization/initialization.go

```diff
@@ -41,6 +41,8 @@ var (
 	ErrAlreadyInitializing          = errors.New("already initializing")
 	ErrCannotResetWhileInitializing = errors.New("cannot reset while initializing")
 	ErrStateMetadataFileMissing     = errors.New("metadata file is missing")
+
+	powDifficultyFunc = shared.PowDifficulty
```

Comment:
Please, let's not use mutable globals. It's one of the sources of data races and prevents running tests in parallel.

How about adding a `powDifficulty func (numLabels uint64) []byte` field to `Initializer struct`, setting it to `shared.PowDifficulty` by default and allowing overriding with `OptionFunc`?

Reply:
It should not be possible to modify this field outside of tests.

The only reason it exists is to be able to test correct behaviour that should only occur for 0.01% of all users.

Reply:
The data race in `gpu-post` (which must be fixed anyway) is not an excuse for introducing another.

If you want to restrict the override to be private and used only inside the package (tests), then you can just make the respective field in the options structure private. What do you think?

Reply:
Even if we fix the data race in `gpu-post` I don't think the library can be used to calculate `PoST` data of two different identities or two batches concurrently. So Initializer should be treated as a singleton anyway.

Nevertheless I removed the global variable and made it a private field.